src/BackgroundTasks/Task/Job.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace ILIAS\BackgroundTasks\Task;
+
+use ILIAS\BackgroundTasks\IO;
+use ILIAS\BackgroundTasks\Task;
+
+/**
+ * Interface Job
+ *
+ * @package ILIAS\BackgroundTasks\Task
+ *
+ *          A Task, which can be run without any interaction with the user such as zipping files or just collecting some data
+ */
+interface Job extends Task {
+
+	/**
+	 * @param \ILIAS\BackgroundTasks\IO $input
+	 * @return \ILIAS\BackgroundTasks\IO
+	 */
+	public function run(IO $input);
+
+
+	/**
+	 * @return bool Returns true iff the job supports giving feedback about the percentage done.
+	 */
+	public function supportsPercentage();
+
+
+	/**
+	 * @return int Returns 0 if !supportsPercentage and the percentage otherwise.
+	 */
+	public function getPercentage();
```

Comment:
That's a good idea! I'll pass the observer to the run method, and add a ping method to the observer.